File: 2019/02.py

import math, copy, re, hashlib
from intcode import IntCode

GOAL = 19690720

def part_1(data):
	a = IntCode(data)
	print(a.data[0])

	print('END OF PART1')
	return

def part_2(data):
	ans = None
	# print(data[1:3])
	for j in range(100):
		if ans is not None:
			break

		for k in range(100):
			data[1] = j
			data[2] = k
			
			i = 0
			d = copy.deepcopy(data)
			d = IntCode(d)
			if d.data[0] == GOAL:
				ans = [j,k]
				break

	print(ans if ans is None else 100 * ans[0] + ans[1])

	print('END OF PART2')
	return 


if __name__ == '__main__':
	with open('02_input') as f:
		data = f.read()
		# data = data.split('\n')
		data = list(map(int, data.split(',')))


	part_1(copy.deepcopy(data))
	part_2(copy.deepcopy(data))
	

File: 2019/02.js
var fs = require('fs');
var { debug, range, drange } = require("../helpers");
var { IntCode } = require("./intcode");

const GOAL = 19690720;

function part1(data) {
	const game = new IntCode(data);
	game.calculate();
	debug(game.data[0]);

	console.log("END OF PART1");
	return;
}

function part2(data) {
	let ans, cnt = 0;
	drange(0, 100)((i, j) => {
		data[1] = i;
		data[2] = j;
		const gameData = Array.from(data);
		const game = new IntCode(gameData);
		game.calculate();
		if (game.data[0] === GOAL) {
			ans = [i, j]
			return true;
		}
	})

	debug(100 * ans[0] + ans[1]);
	console.log("END OF PART2");
	return;
}

function main() {
	let data = fs.readFileSync("02_input").toString("utf-8");
	// data = data.split('\n');
	data = data.split(',').map(a => Number(a));

	if (Array.isArray(data)) {
		part1(Array.from(data));
		part2(Array.from(data));
	} else {
		part1(data);
		part2(data);
	}
}

main();
